src/Components/test/testassets/BasicTestApp/PreserveStateComponent.razor

```diff
@@ -43,27 +44,33 @@
 
     protected override void OnInitialized()
     {
-        if (AppState.TryTakePersistedState(State, out var preserved))
+        if (AppState.TryTake(State, out var preserved))
         {
             _state = Encoding.UTF8.GetString(preserved);
             _restored = true;
         }
         else
         {
             _state = Guid.NewGuid().ToString();
-            AppState.PersistState(State, Encoding.UTF8.GetBytes(_state));
+            AppState.Persist(State, writer => WriteUtf8Bytes(writer, _state));
             _extraState = Guid.NewGuid().ToString();
             if(ExtraState != null)
             {
                 _extraStateAvailable = true;
-                AppState.PersistState(ExtraState, Encoding.UTF8.GetBytes(_extraState));
+                AppState.Persist(ExtraState, writer => WriteUtf8Bytes(writer, _extraState));
             }
         }
+
+        static void WriteUtf8Bytes(IBufferWriter<byte> writer, string state)
+        {
+            var bytes = Encoding.UTF8.GetBytes(state);
+            writer.Write(bytes);
```

Comment:
There's no much difference between the `IBufferWriter<byte>` and the `byte []` when it comes to usage.

Reply:
You can use [this overload](https://docs.microsoft.com/en-us/dotnet/api/system.text.encodingextensions.getbytes?view=net-5.0#System_Text_EncodingExtensions_GetBytes_System_Text_Encoding_System_ReadOnlySpan_System_Char__System_Buffers_IBufferWriter_System_Byte__)

Reply:
Noice, this is for a test, but it is good to know that overload exists